src/servers/src/http/authorize.rs

```diff
@@ -58,12 +58,20 @@ pub async fn inner_auth<B>(
 ) -> std::result::Result<Request<B>, Response> {
     // 1. prepare
     let (catalog, schema) = extract_catalog_and_schema(&req);
+    // TODO(ruihang): move this out of auth module
```

Comment:
Agree, that the query context should be constructed at the header of the request handler chain.